llvm/lib/Target/RISCV/RISCVAsmPrinter.cpp

```diff
@@ -172,6 +199,35 @@ void RISCVAsmPrinter::LowerSTATEPOINT(MCStreamer &OutStreamer, StackMaps &SM,
     assert(PatchBytes % NOPBytes == 0 &&
            "Invalid number of NOP bytes requested!");
     emitNops(PatchBytes / NOPBytes);
+  } else {
+    // Lower call target and choose correct opcode
+    const MachineOperand &CallTarget = SOpers.getCallTarget();
+    MCOperand CallTargetMCOp;
+    switch (CallTarget.getType()) {
```

Comment:
Tests for all types?

Reply:
I added tests based on the `Statepoint` tests from `AArch64` since the architecture are similar.